src/tests/CMakeLists.txt

```diff
@@ -53,16 +53,23 @@ find_package(Boost 1.57.0 REQUIRED COMPONENTS unit_test_framework)
   list(APPEND test_cases test_jobtopology)
   list(APPEND test_cases test_dipoledipoleinteraction)
   list(APPEND test_cases test_populationanalysis)
-list(APPEND test_cases test_gaussian)
-list(APPEND test_cases test_orca)
-list(APPEND test_cases test_nwchem)
-list(APPEND test_cases test_dftengine)
-list(APPEND test_cases test_bsecoupling)
-list(APPEND test_cases test_rate_engine)
+  list(APPEND test_cases test_gaussian)
+  list(APPEND test_cases test_orca)
+  list(APPEND test_cases test_nwchem)
+  list(APPEND test_cases test_dftengine)
+  list(APPEND test_cases test_bsecoupling)
+  list(APPEND test_cases test_rate_engine)
+  if(USE_CUDA)
+    list(APPEND test_cases test_eigencuda)
+  endif()
 
   foreach(PROG ${test_cases} )
     add_executable(unit_${PROG} ${PROG}.cc)
-    target_link_libraries(unit_${PROG} votca_xtp ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})
+    target_link_libraries(unit_${PROG} votca_xtp Boost::unit_test_framework)
+    if(USE_CUDA)
```

Comment:
Is this needed for all tests? Or just `test_eigencuda`?

Plus as you link `votca_xtp` against cuda already, is this needed at all?

Reply:
It is need for all test if `use_CUDA` is `True`.
 Also, Even though `votca_xtp` is link against cuda, if I don't include the cuda libraries I get some compilation errors

Reply:
That just means you have to make cuda a `PUBLIC` link dependency (instead of `PRIVATE`).